README.md

```diff
@@ -302,6 +203,117 @@ Applies only to `install-wcp.bash`
+if [[ "$(platform)" == 'ubuntu' ]]; then
+  echo 'do something specific to Ubuntu'
+fi
+```
+
+```bash
+case "_$(platform)" in
+  _alpine)
+    sudo apk update
+    sudo apk add sl
+    ;;
+  _ubuntu)
+    sudo apt-get update
+    sudo apt-get install sl
+    ;;
+  _*)
+    echo "can't install sl on unrecognized platform: \"$(platform)\""
+    ;;
+esac
+```
+
+The tests and other shells scripts may use the `LABKEY_INSTALL_SKIP_MAIN` environment variable when `source`ing this script to allow for setting all of the script's functions without executing them.
+
+Similarly, a general mechanism for skipping "step functions" has been included that allows users to provide the step's name in an environment variable of the shape: `LABKEY_INSTALL_SKIP_<step fn name>_STEP` which will cause the script to.. you guessed it.. skip that step. For example, to skip the intro step function and void printing the LabKey ascii art: `LABKEY_INSTALL_SKIP_INTRO_STEP=1 ./install-labkey.sh` This is accomplished by the `_skip_step()` function which should be included as the first line in any "step functions":
+
+```bash
+function step_example() {
+  if _skip_step "${FUNCNAME[0]/step_/}"; then return 0; fi
+
+  echo 'an example function'
+}
+```
+
+Setting `LABKEY_INSTALL_SKIP_EXAMPLE_STEP=1` would cause the script in the above example to void printing "an example step".
+
+### Writing Tests
+
+This repo uses an "xUnit"-style testing testing tool called [`shunit2`](https://github.com/kward/shunit2). `shunit2` was chosen over `bats` owning to the lack of strict mode support in `bats` when sourcing other scripts and their functions. See also: [Support for "unofficial strict mode"?](https://github.com/sstephenson/bats/issues/171).
+
+`shunit2` uses "assertions" as the currency with which code functionality is purchased. Some common, self explanatory assertions are: `assertEquals`, `assertNull`, and `assertTrue`. Assertions generally follow the format: `<assertionFunction> <message upon failure> <expected results> <actual results>` and in true Bash fashion, mostly operate on strings. E.g.: `assertEquals 'values not equal' 'apple' "$(fn_which_prints_pear)"` would fail assuming `fn_which_prints_pear` would print "pear".
+
+Tests for specific functionality of a given installation "step" can be written to a test script file named after that steps (as with `step_intro_test.sh`). And tests for internal functions can be written to the `internals.sh` script file.
+
+Try to avoid writing tests that verify the functionality of reliable tools like `mkdir`.
+
+### Running tests Locally
+
+An advantage of having a "purely bash" testing framework is the ability to just add the `shunit2` source file into one's repo, so you'll both find it within the `test` directory and excluded from shells script linting in the github actions.
+
+Since a copy of the `shunit2` source code exists in this repo, the tests are self-contained and assuming they have execute permissions, can be run simply as:
```

Comment:
self-contained and > self-contained and,

Reply:
Edited